ts/util/LinkedList.ts

```diff
@@ -224,48 +209,31 @@ export class LinkedList<DataClass> {
   }
 
   /**
-   * Make the list iterable and return the data from the items in the list
+   * An iterator for the list in forward order
    *
-   * @return {{next: Function}}  The object containing the iterator's next() function
+   * @yield {DataClass} The next item in the iteration sequence
    */
-  public [Symbol.iterator](): Iterator<DataClass> {
+  public *[Symbol.iterator](): IterableIterator<DataClass> {
     let current = this.list;
```

Comment:
This needs to be `this.list.next` not just `this.list`.  Th function as it stands never returns any values, as `this.list.data` is always `END`.